test/Events.test.js

```diff
@@ -0,0 +1,324 @@
+  let EVENTS;
+
+  beforeEach(() => {
+    gameState = new GameState();
+    gameState.currentPlayerActions = {
+      END_TURN: {
+        execute: () => true,
+      },
+    };
+    mockUI = {
+      startGame: () => true,
+      startTurn: (player) => false,
+      displayAvailableActions: (actions) => false,
+      prompt: () => true,
+      endTurn: () => true,
+      rollingDice: () => true,
+      rollDiceDisplay: (shouldDisplay) => true,
+      payFineDisplay: (shouldDisplay) => true,
+      endTurnDisplay: (shouldDisplay) => true,
+      diceRollResults: (roll1, roll2) => true,
+      rollNormalDice: () => true,
+      rollJailDice: () => true,
+      caughtSpeeding: () => true,
+      playerMovement: (position) => true,
+      payFine: () => true,
+      passGo: () => true,
+      jail: () => true,
+      unknownAction: () => true
+    };
+    eventBusEmitter = new EventEmitter();
+    EVENTS = require("../entities/Events")(eventBusEmitter, mockUI, gameState);
+    gameState.players = [createPlayer({ name: "player1" })];
+  });
+  afterEach(() => {
+    // Restore the default sandbox here
+    sinon.restore();
+  });
+  it("constructor should return object with associated event listeners", () => {
+    for (property in EVENTS) {
+      expect(typeof EVENTS[property]).to.equal("function", `Invalid property type, should be function for ${property}`);
+    }
+  });
+
+  describe("startGame", () => {
+    it("should emit start turn event", () => {
+      const startTurnSpy = sinon.spy();
+      eventBusEmitter.on("START_TURN", startTurnSpy);
+      EVENTS.START_GAME();
+      expect(startTurnSpy.callCount).to.equal(1);
+    });
+    it("should make ui call", () => {
+      const uiSpy = sinon.spy();
+      mockUI.startGame = uiSpy;
+      EVENTS.START_GAME();
+      expect(uiSpy.callCount).to.equal(1);
+    });
+  });
+  describe("startTurn", () => {
+    it("should reset speeding counter and turn taken flag", () => {
+      gameState.speedingCounter = 1;
+      gameState.turnTaken = true;
+      EVENTS.START_TURN();
+      expect(gameState.speedingCounter).to.equal(0, "Speeding counter should be reset to 0");
+      expect(gameState.turnTaken).to.equal(false, "Turn should not be taken at the start of turn");
+    });
+    it("should emit continue turn event", () => {
+      const continueTurnSpy = sinon.spy();
+      eventBusEmitter.on("CONTINUE_TURN", continueTurnSpy);
+      EVENTS.START_TURN();
+      expect(continueTurnSpy.callCount).to.equal(1);
+    });
+    it("should make ui call", () => {
+      const startTurnUISpy = sinon.spy();
+      mockUI.startTurn = startTurnUISpy;
+      EVENTS.START_TURN();
+      expect(startTurnUISpy.callCount).to.equal(1);
+      expect(startTurnUISpy.getCall(0).args).to.have.deep.members([gameState.players[0]]);
+    });
+  });
+  describe("passGo", () => {
+    it("should increase player cash by 200", () => {
+      expect(gameState.currentPlayer.cash).to.equal(1500, "Initial player cash is wrong");
```

Comment:
I don't think we should be testing this here - the initial cash amount should be part of the Player or GameState unit, no ? We're sort of double-checking it here in that sense, but it shouldn't be PassGo's responsibility